File: 算法/代码随想录算法PDF/00调试源代码/100.相同的树.cpp
/*
 * @lc app=leetcode.cn id=100 lang=cpp
 *
 * [100] 相同的树
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        stack<TreeNode*> st;
        st.push(p);
        st.push(q);
        while (!st.empty()) {
            p = st.top();st.pop();
            q = st.top();st.pop();
            if (!p && !q) continue;
            if (!p || !q || p->val != q->val) {
                return false;
            }
            st.push(p->left);
            st.push(q->left);
            st.push(p->right);
            st.push(q->right);
        }
        return true;
    }
};
// @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/100.相同的树.py
#
# @lc app=leetcode.cn id=100 lang=python3
#
# [100] 相同的树
#

# @lc code=start
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        st = []
        st.append(p)
        st.append(q)
        while st:
            p = st.pop()
            q = st.pop()
            if not p and not q: continue
            if not p or not q or p.val != q.val:
                return False
            st.append(p.left)
            st.append(q.left)
            st.append(p.right)
            st.append(q.right)
        return True
# @lc code=end

